src/cluster.c

```diff
@@ -276,9 +276,9 @@ void restoreCommand(client *c) {
     }
 
     /* Create the key and set the TTL if any */
-    dbAdd(c->db, key, obj);
+    dbAdd(c->db, key, &obj);
```

Comment:
setKey doesn't take a TTL either. I want to improve this in a follow-up to provide something like dbAddWithExpire and detKeyWithExpire to avoid reallocation. It needs some refactoring and I don't want this PR to become larger than what it already is.